### ISSUE TITLE
Would it be possible to handle dark mode differently?

### ISSUE THREAD

**Issue by Kam3leoN** [materializecss/materialize#386]
### Before submitting...

- [X] I have searched for [duplicate or closed issues](https://github.com/materializecss/materialize/issues?utf8=%E2%9C%93&q=is%3Aissue).
- [X] I have read the [CONTRIBUTING document](https://github.com/materializecss/materialize/blob/master/CONTRIBUTING.md) and my issue is following the guidelines.
- [X] I have read the template completely before filling it in.

### Context

The darkmode is not advanced enough in terms of options

### Current Behavior

The dark mode option is perfect but deserves to be deepened

### Expected behavior

If possible, the ideal would be to be able to activate it as a component

Here are some interesting leads and cool options like:
1. Activation or not via the "materialize.init.js" file like the other components.
2. Dark mode with choice between: Dark, Light and Automatic (ex: in a drop-down menu) the automatic would be executed, if selected
  by the user instead of darkmode depending on the time of day (example "darkmode" between 19pm and 7am otherwise "lightmode", it's just one more condition but very practical).
3. Show toast (“Dark”, “Light”, “Automatic” mode “On”, “Off”).
4. Switching from darkmode to lightmode with a wave effect as (https://vuetifyjs.com/)

### Your Environment

MaterializeCSS : 2.0.1 alpha
Chrome : Version 114.0.5735.134 (Build officiel) (64 bits)
Windows 11 on desktop

**Comment by ivanBell199** [comment]
As a random (not advanced) use of the library I can say that I am completely happy with the current implementation. A simple example of JS theme-toggle and a list of CSS variables are fine for me

**Comment by Kam3leoN** [comment]
The idea is to appeal not just to one person but to a wide audience.
I agree with the fact that it is the developer of the site but in the absolute it is finally a component like any other some will use the sidebar others not, hence the idea of ​​making it a component and to give free rein to choice via options. ;)

**Comment by danice** [comment]
I agree with @Jerit3787 , a framework should leave open how to apply the styles. 
For example, about this light / dark , it also exist the prefers-color-scheme , I have not used that but it seems a way for the user to fix its preferences from the browser:

[media feature](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_media_queries/Using_media_queries#targeting_media_features) is used to detect if a user has requested light or dark color themes. A user indicates their preference through an operating system setting (e.g. light or dark mode) or a user agent setting.

If the framework had implemented what you said it will conflict with this option.

**Comment by kenny-nt** [comment]
I create personal wordpress plugin with materialize and it is default set to dark > make all admin go to dark. How to make default theme to light? I check localstorage and no attribute: theme

**Comment by Jerit3787** [comment]
> I create personal wordpress plugin with materialize and it is default set to dark > make all admin go to dark. How to make default theme to light? I check localstorage and no attribute: theme

Hi @kenny-nt, you can view this issue for solutions #483.

**Comment by kenny-nt** [comment]
html tag currently do not have attribute theme and it is still dark as default. I tried to edit and add theme="light" > make body to light. Is it should be light as default!?

**Comment by Jerit3787** [comment]
> html tag currently do not have attribute theme and it is still dark as default. I tried to edit and add theme="light" > make body to light. Is it should be light as default!?

I'm not sure why yours is dark as default. the default theme is light, and you should also be able to switch to dark.

**Comment by kenny-nt** [comment]
``> > html tag currently do not have attribute theme and it is still dark as default. I tried to edit and add theme="light" > make body to light. Is it should be light as default!?
> 
> I'm not sure why your default theme is dark. The default theme for materialize is light, and you should also be able to switch to dark. If you could provide how the implementations on your side would be appreciated.

I just enqueue materialize into an option page of wordpress (wp-admin_ . I dont setup html tag, the default html tag of wordpress admin backend. I checked html tag:
```
<html class="wp-toolbar wf-poppins-n7-inactive wf-poppins-n4-inactive wf-dancingscript-n4-inactive wf-inactive" lang="en-US">
<body>
.....
</body>
</html>
```
But the page is dark mode... It is not happened with 2.0.4

**Comment by wuda-io** [comment]
The default theme depends now on the system preferences. If you have set your System to dark mode then Materialize will also render in dark. If you have default light mode, then it will render light.

You can override the theme via Javascript. The documentation needs to be updated. And also all the old theme variables are deprecated. We use now directly the M3 vars, because they were already mix up and it was cleaner to use them directly. There is still a lot of work to do, so please contribute. Every support is very welcome! Thanks.
